src/components/modals/AddUserModal.jsx

```diff
@@ -0,0 +1,240 @@
+    setState({
+      ...state,
+      submitted: true
+    });
+    const validForm = state.displayNameValid && state.emailNameValid;
+    if (!validForm) return;
+
+    const user = {
+      displayName: state.displayName,
+      emailPreference: state.emailPreference,
+      roles: state.updatedRoles,
+      email: state.email
+    };
+
+    const createdUser = await User.create(user);
+
+    setState({
+      ...state,
+      emailValid: createdUser
+    });
+    event.preventDefault();
+
+    if (state.emailValid !== false) {
+      props.onOKRequest('addUser');
+    }
+  };
+
+  const closeHandler = () => {
+    props.onCloseRequest('addUser');
+  };
+
+  const afterOpenHandler = () => {
+    props.onAfterOpen('addUser');
+  };
+
+  const emailPreferenceChanged = (e) => {
+    const checkState = e.target.checked;
+    setState({
+      ...state,
+      emailPreference: !checkState
+    });
+  };
+
+  const adminChanged = (e) => {
+    const checkState = e.target.checked;
+    let newRoles = [researcherRole];
+    if (checkState) {
+      newRoles = _.concat(state.updatedRoles, adminRole);
+    } else {
+      newRoles = _.filter(state.updatedRoles, (r) => r.roleId !== adminRole.roleId);
+    }
+    setState({
+      ...state,
+      updatedRoles: newRoles
+    });
+  };
+
+  const handleChange = (e) => {
+    const name = e.target.name;
+    const validName = name + 'Valid';
+    setState({
+      ...state,
+      [name]: e.target.value,
+      [validName]: e.currentTarget.validity.valid
+    });
+  };
+
+  const formChange = () => {
+    setState((prev) => ({
+      ...prev,
+      invalidForm: prev.displayNameValid && prev.emailValid
+    }));
+  };
+
+  const isAdmin = () => {
+    const admins = _.filter(state.updatedRoles, _.matches(adminRole));
+    return !_.isEmpty(admins);
+  };
+
+
+  const { displayName, email, displayNameValid, emailValid } = state;
+  const validForm = displayNameValid && emailValid;
+
+  return (
+    <BaseModal
+      id='addUserModal'
+      showModal={props.showModal}
+      disableOkBtn={!validForm}
+      onRequestClose={closeHandler}
+      onAfterOpen={afterOpenHandler}
+      imgSrc={addUserIcon}
+      color='common'
+      title='Add User'
+      description='Catalog a new User in the system'
+      action={{ label: 'Add', handler: OKHandler }}
+    >
+      <form className='form-horizontal css-form' name='userForm' encType='multipart/form-data' onChange={formChange}>
+        <div className='form-group first-form-group'>
+          <label id='lbl_name' className='col-lg-3 col-md-3 col-sm-3 col-xs-4 control-label common-color'>Name</label>
+          <div className='col-lg-9 col-md-9 col-sm-9 col-xs-8'>
+            <input
+              type='text'
+              name='displayName'
+              id='txt_name'
+              className='form-control col-lg-12 vote-input'
+              placeholder='User name'
+              required={true}
+              value={displayName}
+              autoFocus
```

Comment:
Do we need to provide a value for this since `autoFocus` isn't a variable anywhere in this component?
```suggestion
              autoFocus={true}
```